### PR TITLE
limit max number of write ops

### PR DESCRIPTION
**Pull request by msmouse**
### Description
Add some safety guard, just in case.

### CODE REVIEW

File: aptos-move/aptos-vm-types/src/change_set.rs [hunk 1]
```diff
@@ -159,6 +159,12 @@ impl VMChangeSet {
             .chain(self.aggregator_write_set.iter())
     }
 
+    pub fn num_write_ops(&self) -> usize {
+        self.resource_write_set.len()
+            + self.module_write_set().len()
+            + self.aggregator_v1_write_set().len()
```
Comment (gelash) [inline aptos-move/aptos-vm-types/src/change_set.rs:165]: shouldn't we be adding delta set as well, those will become writes too (not that it matters for V1, but logically seems true?)

Comment (msmouse) [reply aptos-move/aptos-vm-types/src/change_set.rs:165]: Oh.. actually this is after everything is materialized, there shouldn't be deltas.

Comment (gelash) [reply aptos-move/aptos-vm-types/src/change_set.rs:165]: Are you sure? we could assert in that case, but I thought checker happens in the session and then it's likely it wouldn't be materialized @georgemitenkov 

delta gas is for sure smt we need to do :-)

Comment (gelash) [reply aptos-move/aptos-vm-types/src/change_set.rs:165]: this can't be after materialization because then you wouldn't be able to abort the transaction?

Comment (msmouse) [reply aptos-move/aptos-vm-types/src/change_set.rs:165]: K I'm wrong. It's after user session before epilogue.
https://github.com/aptos-labs/aptos-core/blob/48aac6d3d9f654198defec4b4b0eeb034abd40e1/aptos-move/aptos-vm/src/move_vm_ext/session.rs#L385

Comment (georgemitenkov) [reply aptos-move/aptos-vm-types/src/change_set.rs:165]: We should have a limit on deltas probably, but I think we can do that as a part of parallel execution gas (also behaviour between V1 and V2 is a bit different)? @gelash

File: aptos-move/aptos-vm-types/src/storage.rs [hunk 2]
```diff
@@ -278,6 +278,7 @@ impl StoragePricing {
 #[derive(Clone, Debug)]
 pub struct ChangeSetConfigs {
     gas_feature_version: u64,
+    max_write_ops_per_transaction: u64,
```
Comment (zekun000) [inline aptos-move/aptos-vm-types/src/storage.rs:281]: can we move it to the end to align with below?

File: aptos-move/aptos-vm-types/src/storage.rs [hunk 3]
```diff
@@ -353,6 +357,12 @@ impl CheckChangeSet for ChangeSetConfigs {
     fn check_change_set(&self, change_set: &VMChangeSet) -> Result<(), VMStatus> {
         const ERR: StatusCode = StatusCode::STORAGE_WRITE_LIMIT_REACHED;
 
+        if self.max_write_ops_per_transaction != 0
```
Comment (zekun000) [inline aptos-move/aptos-vm-types/src/storage.rs:360]: why != 0?

Comment (msmouse) [reply aptos-move/aptos-vm-types/src/storage.rs:360]: before gas schedule is updated (and gas feature is enabled) the gas parameter will read 0

File: aptos-move/aptos-vm-types/src/storage.rs (hunk 2 above)
Comment (msmouse) [reply aptos-move/aptos-vm-types/src/storage.rs:281]: okay...

File: aptos-move/aptos-vm-types/src/storage.rs (hunk 3 above)
Comment (zekun000) [reply aptos-move/aptos-vm-types/src/storage.rs:360]: I thought you put u64::MAX there?

Comment (msmouse) [reply aptos-move/aptos-vm-types/src/storage.rs:360]: no see `fn from_gas_params()`

**Review by grao1991 — APPROVED** [review]

**Review by zekun000 — APPROVED** [review]

**Pull request merged**